### DISCUSSION TITLE
Conceptual question on using QUDT to add measurement specificity to an ontology

### DISCUSSION THREAD

**Discussion by ijdickinson** [qudt/qudt-public-repo#670] [Q&A] [answered]
Apologies if this is a basic question, but I've been looking at the QUDT docs and I can't quite get my head around this.

I want my ontology to express that an `ex:area` property connects a resource of `rdf:type` habitat to a value that has units denoting an area (but not, say, a length). As a weaker version of this, a habitat should have an area denoted in hectares (though in truth we may get input data with units of m^2). 

My best approximation so far is:

```turtle
ex:Habitat 
  a owl:Class ;
  rdfs:subClassOf [
    a owl:Restriction ;
    owl:onProperty ex:area ;
    owl:someValuesFrom [
      a owl:Restriction ;
      owl:onProperty qudt:unit ;
      owl:exactCardinality 1 ;
      owl:allValuesFrom unit:ha
    ]
  ]
.
```

But this seems unnecessarily complicated. Am I missing a more simple or direct way to express this?

**Comment by steveraysteveray** [comment, marked as answer]
I have a few suggestions for you.

1. It's important to recognize the distinction between your relation "area" and the QuantityKind quantitykind:Area. To avoid confusion, I suggest you name your property/relation ex:hasArea.
2. Secondly, what you are saying in English is "My Habitat class has a property 'area' (or better, hasArea) that is an object that has a relation to a numerical value and a second relation to a qudt:Unit instance that I want to constrain." So you need to first define the structure of your Area class (to at least have a numerical value and an associated Unit). The quickest way to do this is to define your own subclass of qudt:Quantity, which has the necessary relations already defined, and then constrain the values of some of those relations. 


```
ex:Habitat 
  a owl:Class ;
  rdfs:subClassOf [
    a owl:Restriction ;
    owl:onProperty ex:hasArea ;
    owl:allValuesFrom ex:HabitatArea ;
    ] ;
.
ex:HabitatArea 
  a owl:Class ;
  rdfs:subClassOf qudt:Quantity ;
    rdfs:subClassOf [
        a owl:Restriction ;
        owl:onProperty qudt:hasQuantityKind ;
        owl:hasValue quantitykind:Area ;
    ] ;
.

```
Of course, you could just nest the HabitatArea inside the definition of Habitat, but the bottom line is that you aren't constraining the value of an instance of Habitat, you are constraining the instance of the quantity that the Habitat instance is pointing to. So you will always be forced to have a "double-hop" in your constraint, no matter how you structure it.

Also, you might find our [User Guide](https://github.com/qudt/qudt-public-repo/wiki/User-Guide-for-QUDT) helpful.

One final note - have you considered using SHACL instead of OWL? I find it a bit clearer when specifying constraints like this.
